test/standalone/Vegetation/plant_hydraulics_test.jl

```diff
@@ -303,31 +306,31 @@ for FT in (Float32, Float64)
                                     ),
                                 ) .*
                                 ClimaLand.Canopy.PlantHydraulics.root_distribution.(
-                                    root_depths,
+                                    root_depths[j],
                                     plant_hydraulics.parameters.rooting_depth,
-                                ) .* (
-                                    vcat(root_depths, [0.0])[2:end] -
-                                    vcat(root_depths, [0.0])[1:(end - 1)]
-                                ),
-                            ) * AI[:stem]
-                    else
-                        fa =
-                            water_flux(
-                                plant_hydraulics.compartment_midpoints[i - 1],
-                                plant_hydraulics.compartment_midpoints[i],
-                                Y[i - 1],
-                                Y[i],
-                                PlantHydraulics.hydraulic_conductivity(
-                                    conductivity_model,
+                                ) .* FT(
+                                    j != length(root_depths) ?
+                                    root_depths[j + 1] : 0.0,
+                                ) .* root_depths[j] .* AI[:stem]
+                        else
+                            fa .+=
+                                water_flux(
+                                    plant_hydraulics.compartment_midpoints[i - 1],
+                                    plant_hydraulics.compartment_midpoints[i],
                                     Y[i - 1],
-                                ),
-                                PlantHydraulics.hydraulic_conductivity(
-                                    conductivity_model,
                                     Y[i],
-                                ),
-                            ) * AI[plant_hydraulics.compartment_labels[i]]
+                                    PlantHydraulics.hydraulic_conductivity(
+                                        conductivity_model,
+                                        Y[i - 1],
+                                    ),
+                                    PlantHydraulics.hydraulic_conductivity(
+                                        conductivity_model,
+                                        Y[i],
+                                    ),
+                                ) * AI[plant_hydraulics.compartment_labels[i]]
+                        end
+                        F[i] = mean(fa .- T0A)
```

Comment:
Let's discuss this. I would have expected `F` to be a `Field` here too

Reply:
I believe that in this test, we solve for the steady state of a vector-valued quantity using a root solver. We then set the initial condition of the field to the value of this vector. This seemed easier than getting the root solver to play with Fields, though maybe that is easy enough to do

Reply:
@Sbozzolo I changed this test to not take the mean, and instead follow the original behavior but with fields. My implementation of this change is abnormal (see last commit). Could we discuss?

Reply:
It's a little abnormal, but I think it's fine! Thanks for the comments in the commit message too